File: Phase#1/Ramadan_Marathon/contest#5/e.py
#!/usr/bin/python3

if __name__ == '__main__':
    t = 1
    # t = int(input())

    for _ in range(t):
        n,k = map(int, input().split())
        freqA = [0]*100010
        freqB = [0]*100010

        arrA = list(map(int, input().split()))
        arrB = list(map(int, input().split()))

        def getFreq(arr, freq):
            for i in arr:
                freq[i] += 1

        getFreq(arrA, freqA)
        getFreq(arrB, freqB)

        def countPairs(freq):
            cnt = 0
            for i in arrA:
                freq[i] -= 1
                if k-i > 0:
                    cnt += freq[k-i]
            return cnt

        cnt1 = countPairs(freqA)
        cnt2 = countPairs(freqB)
        if cnt1 == cnt2:
            print("Draw")
        elif cnt1 > cnt2:
            print("Mahmoud")
        else:
            print("Bashar")

File: Phase#1/Ramadan_Marathon/contest#5/e.cpp
#include <iostream>

#define endl "\n"

using namespace std;

void fastIO() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
}

int main() {
    fastIO();
    int n, k;
    cin >> n >> k;
    int arrA[n];
    int arrB[n];
    int freqA[100010];
    int freqB[100010];
    for (int i = 1; i <= n; i++) {
        cin >> arrA[i];
        freqA[arrA[i]]++;
    }
    for (int i = 1; i <= n; i++) {
        cin >> arrB[i];
        freqB[arrB[i]]++;
    }
    int cnt1 = 0;
    int cnt2 = 0;

    for (int i = n; i > 1; i--) {
        freqA[arrA[i]]--;
        if (k - arrA[i] > 0) {
            cnt1 += freqB[k - arrA[i]];
        }
    }
    for (int i = n; i > 1; i--) {
        freqB[arrB[i]]--;
        if (k - arrB[i] > 0) {
            cnt2 += freqA[k - arrB[i]];
        }
    }

    cout << cnt1 << " " << cnt2 << endl;
    if (cnt1 > cnt2) {
        cout << "Mahmoud" << endl;
    } else if (cnt1 < cnt2) {
        cout << "Bashar" << endl;
    } else {
        cout << "Draw" << endl;
    }

}
